core/src/main/scala/io/finch/request/package.scala

```diff
@@ -162,6 +180,8 @@ package object request {
       def apply[Req](req: Req)(implicit ev: Req => HttpRequest) =
         try number.toFuture
         catch { case _: IllegalArgumentException => ValidationFailed(param, rule).toFutureException }
+
+      override def toString: String = s"param: $param"
```

Comment:
Cool! Feel free to start an issue i.e., "Composable Validation Rules" if you want to discuss either  design or implementation in details. Otherwise, a PR would be a great reason to start a discussion.